HW0/P4/P4.py

```diff
@@ -40,37 +42,132 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    C_partial = np.matrix([
+                            [1.0/rx,0,0],
+                            [0,1.0/ry,0],
+                            [0,0,1.0/rz]
+                         ])
+    s_measured = np.loadtxt('P4_measurements.txt', 
+             delimiter = ',', unpack = True)
+
+    s_measured = np.matrix(s_measured)
+    (cols, rows) = s_measured.shape
+
+    measured_output = np.zeros([cols,rows])
+    measured_output = np.asmatrix(measured_output)
+
+    for i in range(0,rows):
+        measured_output[:,i] = C_partial * s_measured[:,i]
+
+    measured_output = np.array(measured_output)
+
+    ax.plot(measured_output[0], measured_output[1], measured_output[2],
+             '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
-
-    # A = ?
-    # a = ?
-    # s = ?
+    drag_per_dt = 1 - c*dt
+    A = np.matrix([
+                    [1,0,0,dt,0,0],
+                    [0,1,0,0,dt,0],
+                    [0,0,1,0,0,dt],
+                    [0,0,0,drag_per_dt,0,0],
+                    [0,0,0,0,drag_per_dt,0],
+                    [0,0,0,0,0,drag_per_dt]
+                 ])
+    #print A
+    a = np.matrix([
+                    [0],
+                    [0],
+                    [0],
+                    [0],
+                    [0],
+                    [g*dt]
+                 ])
+    #print a
 
     # Initial conditions for s0
-    # Compute the rest of sk using Eq (1)
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    s_0 = np.matrix([
+                    [s_true[0][0]],
+                    [s_true[1][0]], 
+                    [s_true[2][0]], 
+                    [s_true[3][0]],
+                    [s_true[4][0]],
+                    [s_true[5][0]]
+                 ])
+    #print s_column
+
+    s = np.zeros([6,K])
+    s = np.asmatrix(s)
+
+    # construct s using Eq (1)
+    s_column = s_0
+    s[:,0] = s_column
+    for k in range(1,K):
+        s[:,k] = A*s_column + a
+        s_column = s[:,k]
+
+    s = np.array(s)
+    ax.plot(s[0], s[1], s[2],
+             '-k', label='Blind trajectory')
```

Comment:
This applies throughout, but I would add more comments (maybe referencing equation numbers) describing what everything is doing.